cmd/userns.go

```diff
@@ -91,6 +98,33 @@ func maybeReexecUsingUserNamespace(uidmap string, useNewuidmap bool, gidmap stri
 		}
 	}
 
+	// Log the ID maps we were started with.
+	UIDs, GIDs, err := unshare.GetHostIDMappings("")
+	if err != nil {
+		klog.Fatalf("Error reading current ID mappings: %v\n", err)
+	}
+	isNodeDefaultMapping := func(m []specs.LinuxIDMapping) bool {
+		return len(m) == 1 && m[0].ContainerID == 0 && m[0].HostID == 0 && m[0].Size == 0xffffffff
+	}
+	if !isNodeDefaultMapping(UIDs) || !isNodeDefaultMapping(GIDs) {
+		uidMap, gidMap := "[", "["
+		for i, entry := range UIDs {
+			if i > 0 {
+				uidMap += ", "
+			}
+			uidMap += fmt.Sprintf("(%d:%d:%d)", entry.ContainerID, entry.HostID, entry.Size)
+		}
+		for i, entry := range GIDs {
+			if i > 0 {
+				gidMap += ", "
+			}
+			gidMap += fmt.Sprintf("(%d:%d:%d)", entry.ContainerID, entry.HostID, entry.Size)
+		}
+		uidMap += "]"
+		gidMap += "]"
+		klog.Infof("Started in kernel user namespace with UID map %s and GID map %s.", uidMap, gidMap)
```

Comment:
Is this something that we always want to log, log at debug level (5), or somewhere in between (2)?

Reply:
I don't personally have a preference.  It's mainly there so that we can test for it, but the test can also adjust the logging level.  Do you have a preference?